File: Project_Euler/2.rb
#Making Fibonacii generator:
fibonacci = Enumerator.new do |yielder|
    a,b=1,1
    while a
        yielder.yield a
        a,b=b,a+b
    end
end
puts fibonacci.take_while{|x| x<=4_000_000}.select{|x| x%2==0}.inject(:+)
=begin
#crap approach
arr=Array.new(4000003,0)
arr[1],arr[2]=1,2
sum=2
for x in 3..4000001
arr[x]=arr[x-1]+arr[x-2]
sum+=arr[x] if arr[x]%2==0
end
p sum
=end

File: Project_Euler/2.cpp
#include<iostream>
#include<stdio.h>
using namespace std;

int main()
{
	long unsigned int arr[4000003], i;
	arr[0]=0; arr[1]=1; arr[2]=2;
	long unsigned int sum=2;
	for(i=3; i<=4000001; i++)
	{
		arr[i]=arr[i-1]+arr[i-2];
		if(arr[i]%2==0)sum+=arr[i];
	}
	cout<<sum;
	return 0;
}
